tests/unit/views/SampleReport.spec.js

```diff
@@ -0,0 +1,260 @@
+  const actual = await importOriginal()
+  return {
+    ...actual,
+    downloadBlob: vi.fn(), // mock the downloadBlob method
+  }
+})
+vi.mock('@/lib/reports/KinnexReport.js', async (importOriginal) => {
+  const actual = await importOriginal()
+  return {
+    ...actual,
+    fetchFunction: mocks.fetchFunction,
+  }
+})
+
+describe('SampleReport', () => {
+  let rootStore, get, ss_get
+  const buildWrapper = () => {
+    return mount(SampleReport)
+  }
+
+  beforeEach(() => {
+    /*Creates a fresh pinia instance and make it active so it's automatically picked
+        up by any useStore() call without having to pass it to it for e.g `useStore(pinia)`*/
+    const pinia = createPinia()
+    setActivePinia(pinia)
+
+    // Mock the root store and its API
+    rootStore = useRootStore()
+    get = vi.fn()
+    rootStore.api.traction.pacbio.requests.get = get
+
+    ss_get = vi.fn()
+    rootStore.api.sequencescape.studies.get = ss_get
+  })
+
+  describe('report type', () => {
+    it('contains a list of report types', () => {
+      const expectedOptions = [...Object.values(Reports).map((r) => r.text)]
+      const wrapper = buildWrapper()
+      expect(
+        wrapper
+          .find('[data-type=report-type]')
+          .findAll('option')
+          .map((element) => element.text()),
+      ).toEqual(expectedOptions)
+    })
+
+    it('resets the sample list when the report type changes', async () => {
+      const wrapper = buildWrapper()
+      wrapper.vm.samples = [{ request_id: 1 }, { request_id: 2 }]
+      expect(wrapper.vm.samples.length).toBe(2)
+
+      const select = wrapper.find('[data-type=report-type]')
+      await select.setValue('mock-report-type')
+
+      expect(wrapper.vm.samples.length).toBe(0)
+    })
+  })
+
+  describe('sample input', () => {
+    it('adds a sample when the input is not empty', async () => {
+      const wrapper = buildWrapper()
+      vi.mocked(fetchFunction).mockResolvedValue({
+        data: [{ request_id: 1 }],
+        errors: {},
+      })
+
+      const sampleInput = wrapper.find('#sampleInput')
+      await sampleInput.setValue('sample1')
+      await wrapper.find('#searchSamples').trigger('click')
+      await flushPromises()
+
+      expect(wrapper.vm.samples).toEqual([{ request_id: 1 }])
+    })
+
+    it('is disabled when the report type is not set', async () => {
+      const wrapper = buildWrapper()
+      wrapper.vm.report_type = null
+      await wrapper.vm.$nextTick()
+
+      expect(wrapper.find('#sampleInput').element.disabled).toBe(true)
+      expect(wrapper.find('#searchSamples').element.disabled).toBe(true)
+    })
+
+    it('shows the correct error message when no samples are found', async () => {
+      const wrapper = buildWrapper()
+      vi.mocked(fetchFunction).mockResolvedValue({
+        data: [],
+        errors: { message: 'Internal server error', type: 'danger' },
+      })
+
+      const sampleInput = wrapper.find('#sampleInput')
```

Comment:
I would personally use data-attribute or other rather than id.

Reply:
Updated in [be96327](https://github.com/sanger/traction-ui/pull/2370/commits/be963278088eea3cb21330f9fd17dbc4d71cb13d)